File: Level1/핸드폰 번호 가리기.py
def solution(phone_number):
    answer=''
    cnt=0
    
    #거꾸로
    for i in range(len(phone_number)-1,-1,-1):
        #4자리만 번호로 보여야하므로 cnt로 구분 
        if cnt<4: 
            answer+=phone_number[i]
            cnt+=1
        else:
            answer+='*'
    
    #다시 거꾸로
    return answer[::-1] 

File: Level1/핸드폰 번호 가리기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string phone_number) {
    string answer = "";
    string tmp = "";
    int cnt = 0;
    
    //거꾸로 탐색
    for (int i = phone_number.length() - 1; i >= 0; --i) {
        if (cnt < 4) {
            tmp += phone_number[i];
            cnt++;
        }
        else
            tmp += '*';
    }

    //다시 거꾸로
    for (int i = tmp.length() - 1; i >= 0; --i)
        answer += tmp[i];

    //반환
    return answer;
}